egs/librispeech/ASR/pruned_transducer_stateless2/scaling.py

```diff
@@ -111,6 +111,74 @@ def backward(
         return x_grad - neg_delta_grad, None, None, None, None, None, None
 
 
+class GradientFilterFunction(torch.autograd.Function):
+    @staticmethod
+    def forward(
+        ctx,
+        x: Tensor,
+        batch_dim: int,  # e.g., 1
+        threshold: float,  # e.g., 10.0
+        *params: nn.Parameter,  # module parameters
+    ):
+        if x.requires_grad:
+            if batch_dim < 0:
+                batch_dim += x.ndim
+            ctx.batch_dim = batch_dim
+            ctx.threshold = threshold
+        return x, *params
+
+    @staticmethod
+    def backward(
+        ctx,
+        x_grad: Tensor,
+        *param_grads: nn.Parameter,
```

Comment:
these will be Tensor, not nn.Parameter.

Reply:
... also the type would probably be Tuple[nn.Parameter], but I'm not 100% sure, you could perhaps just omit the typing information (or check).

Reply:
Referring to https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values, the type of `*param_grads` is `Tensor` instead of `Tuple[Tensor]`.